plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+        type: bool
+      rewrite_source_mac:
+        description:
+        - Whether to rewrite the source MAC address.
+        type: bool
+      anycast:
+        description:
+        - Indicates if anycast is enabled.
+        type: bool
+      config_static_mac:
+        description:
+        - Indicates if static MAC configuration is enabled.
+        type: bool
+      is_backup_redirect_ip:
+        description:
+        - Indicates if it is a backup redirect IP.
+        type: bool
+      load_balance_hashing:
+        description:
+        - Load balancing hashing method.
+        type: str
+        choices: ["sourceDestinationAndProtocol", "sourceIP", "destinationIP"]
+      pod_aware_redirection:
+        description:
+        - Indicates if pod-aware redirection is enabled.
+        type: bool
+      resilient_hashing:
+        description:
+        - Indicates if resilient hashing is enabled.
+        type: bool
+      tag_based_sorting:
+        description:
+        - Indicates if tag-based sorting is enabled.
+        type: bool
+      min_threshold:
+        description:
+        - Minimum threshold value for redirect.
+        type: int
+      max_threshold:
+        description:
+        - Maximum threshold value for redirect.
+        type: int
+      threshold_down_action:
+        description:
+        - Action to take when the threshold is down.
+        type: str
+        choices: [ permit, deny, bypass ]
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a service device cluster with one arm
+  cisco.mso.ndo_service_device_cluster:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_device_template
+    name: device1
+    device_mode: layer3
+    device_type: firewall
+    interface_properties:
+      - name: "interface1"
+        type: "l3out"
+        interface:
+          name: ansible_test_epg
+          template: ansible_template
+          schema: ansible_test
+        ipsla_monitoring_policy:
+          name: ansible_test_ipsla_monitoring_policy
+          template: ansible_tenant_template
+        qos_policy:
+          name: ansible_custom_qos_policy
+          template: ansible_tenant_template
+        preferred_group: true
+        rewrite_source_mac: false
+        config_static_mac: false
+        is_backup_redirect_ip: true
+        load_balance_hashing: sourceIP
+        pod_aware_redirection: false
+        resilient_hashing: true
+        tag_based_sorting: false
+        min_threshold: 10
+        max_threshold: 100
+        threshold_down_action: permit
+    state: present
+  register: add_device1
+
+- name: Update the service device cluster to three arm
```

Comment:
should we keep the terminology mentioned in the UI?
```suggestion
- name: Update the service device cluster to advanced
```